h/activity/query.py

```diff
@@ -140,9 +143,21 @@ def aggregations_for(query):
 
 
 @newrelic.agent.function_trace()
+def fetch_annotations(session, ids, reply_ids):
+    def load_documents(query):
+        return query.options(subqueryload(Annotation.document))
+
+    annotations = storage.fetch_ordered_annotations(
```

Comment:
It's worth keeping in mind that `fetch_ordered_annotations` does a bunch of work in Python that can be done by Postgres if the annotations are ordered by a column in the database. This may turn out to be irrelevant -- just flagging it up.